### ISSUE TITLE
Target [Illuminate\Contracts\Auth\Access\Gate] is not instantiable.

### ISSUE THREAD

**Issue by rajanrawal** [ollieread/multiauth#107]
Laravel 5.1.16

php artisan clear compiled

composer updated 

composer dump-autoload -o

All tried but getting

App crashing. Throwing error:
```
BindingResolutionException in Container.php line 744: 
Target [Illuminate\Contracts\Auth\Access\Gate] is not instantiable.
```

**Comment by ollieread** [comment]
Something is trying to instantiate the Gate interface. I'll check through the code but it may be worth checking your own to see what's going on.

**Comment by ollieread** [comment]
Assuming that you're using the Laravel 5 branch, I've searched the entire codebase for the term gate, the only time it appears, ever, is in a comment that actually says `aggregate`.

**Comment by rajanrawal** [comment]
@ollieread I am really sorry. Yes I am using 5.1.16. I enjoyed you package for 4.2 a lot. But there is nothing out there for laravel >= 5.1.11 + which work nice as yours was!

**Comment by lucas-macedo** [comment]
I have the same problem...
Target [Illuminate\Contracts\Auth\Access\Gate] is not instantiable. 
Using Laravel 5.1

**Comment by ollieread** [comment]
Can one of you post a stacktrace so that I can exactly what's going on?

**Comment by jbrooksuk** [comment]
Have you added `Gate` as an alias to `./config/app.php`?

```
'Gate'      => 'Illuminate\Support\Facades\Gate',
```

**Comment by rajanrawal** [comment]
Yes of course we did it. Though it was not working.

On Wed, Sep 16, 2015 at 1:06 PM, James Brooks <notifications@github.com>
wrote:

> Have you added Gate as an alias to ./config/app.php?
>
> 'Gate'      => 'Illuminate\Support\Facades\Gate',
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ollieread/multiauth/issues/107#issuecomment-140656327>
> .
>



-- 
Intensity works... Always...

Thank You & Regards,
Rajan Y. Rawal
+91 903 315 5300

**Comment by jbrooksuk** [comment]
What about the last two bits I suggested?

**Comment by rajanrawal** [comment]
No workout. 

Here is the stacktrace

`BindingResolutionException in Container.php line 744:
Target [Illuminate\Contracts\Auth\Access\Gate] is not instantiable.
in Container.php line 744
at Container->build('Illuminate\Contracts\Auth\Access\Gate', array()) in Container.php line 626
at Container->make('Illuminate\Contracts\Auth\Access\Gate', array()) in Application.php line 674
at Application->make('Illuminate\Contracts\Auth\Access\Gate') in Container.php line 572
at Container->addDependencyForCallParameter(object(ReflectionParameter), array(), array()) in Container.php line 532
at Container->getMethodDependencies(array(object(AuthServiceProvider), 'boot'), array()) in Container.php line 500
at Container->call(array(object(AuthServiceProvider), 'boot')) in Application.php line 734
at Application->bootProvider(object(AuthServiceProvider)) in Application.php line 717
at Application->Illuminate\Foundation\{closure}(object(AuthServiceProvider), '14')
at array_walk(array(object(EventServiceProvider), object(RoutingServiceProvider), object(ControllerServiceProvider), object(CookieServiceProvider), object(DatabaseServiceProvider), object(EncryptionServiceProvider), object(FilesystemServiceProvider), object(FormRequestServiceProvider), object(FoundationServiceProvider), object(PaginationServiceProvider), object(SessionServiceProvider), object(ValidationServiceProvider), object(ViewServiceProvider), object(AppServiceProvider), object(AuthServiceProvider), object(EventServiceProvider), object(RouteServiceProvider), object(MultiauthServiceProvider), object(SluggableServiceProvider)), object(Closure)) in Application.php line 718
at Application->boot() in BootProviders.php line 17
at BootProviders->bootstrap(object(Application)) in Application.php line 203
at Application->bootstrapWith(array('Illuminate\Foundation\Bootstrap\DetectEnvironment', 'Illuminate\Foundation\Bootstrap\LoadConfiguration', 'Illuminate\Foundation\Bootstrap\ConfigureLogging', 'Illuminate\Foundation\Bootstrap\HandleExceptions', 'Illuminate\Foundation\Bootstrap\RegisterFacades', 'Illuminate\Foundation\Bootstrap\RegisterProviders', 'Illuminate\Foundation\Bootstrap\BootProviders')) in Kernel.php line 222
at Kernel->bootstrap() in Kernel.php line 117
at Kernel->sendRequestThroughRouter(object(Request)) in Kernel.php line 87
at Kernel->handle(object(Request)) in index.php line 54`

**Comment by jbrooksuk** [comment]
Have you ran `composer update` **before** adding the alias?

**Comment by rajanrawal** [comment]
Yes all done, basically there is forum question that one asked on laravel forum and the steps you said were mentioned there

https://laracasts.com/discuss/channels/laravel/l51-unable-to-fetch-the-gate-facade-through-composer?page=0

**Comment by jbrooksuk** [comment]
I already suggested deleting everything in `bootstrap/cache` ;)

**Comment by ollieread** [comment]
So this isn't a multiauth issue?

**Comment by jbrooksuk** [comment]
No, I'm pretty sure that it's not.

**Comment by ollieread** [comment]
Ahh excellent, I couldn't recreate this anyway!

**Comment by rajanrawal** [comment]
@jbrooksuk dude what laravel version you are using? Are you on 5.1.16? The latest one?

**Comment by rajanrawal** [comment]
@ollieread of course close, but let test it once. @jbrooksuk I cleaned bootstrap cache as well. Its same here still

**Comment by jbrooksuk** [comment]
@rajanrawal  Yeah, the latest.

**Comment by rajanrawal** [comment]
I tested everything as it was informed here, its still not working at all and throwing the same error. Anyway thank you.

**Comment by Schnapse** [comment]
I'm having the same error as @rajanrawal. I'm currently using the version 5.1.17 of Laravel.

```
php artisan --version
Laravel Framework version 5.1.17 (LTS)
```

I added sboo/multiauth to composer.json : 

```"sboo/multiauth" : "4.0.*"```

Then I launched ```composer update``` command. ```sboo/multiauth v4.0.9``` is installed successfully and everything is going fine, no error at this moment. After that, I ran ```composer du```.

But when I replace ''Illuminate\Auth\AuthServiceProvider' by 'Ollieread\Multiauth\MultiauthServiceProvider' and 'Illuminate\Auth\Passwords\PasswordResetServiceProvider' by 'Ollieread\Multiauth\Passwords\PasswordResetServiceProvider', I'm getting this error : 

```
BindingResolutionException in Container.php line 744:
Target [Illuminate\Contracts\Auth\Access\Gate] is not instantiable.
```

As I read on differents issues, I cleared cache by deletting ```boostrap/cache/*``` folder by still having the error. Am I doing something wrong ?
